src/VisualStudio/Core/Def/Implementation/TableDataSource/DiagnosticTableItem.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
+{
+    internal sealed class DiagnosticTableItem : TableItem
+    {
+        public readonly DiagnosticData Data;
+
+        public DiagnosticTableItem(Workspace workspace, SharedInfoCache cache, DiagnosticData data)
+            : base(workspace, cache)
+        {
+            Contract.ThrowIfNull(data);
+            Data = data;
+        }
+
+        public override TableItem WithCache(SharedInfoCache cache)
+            => new DiagnosticTableItem(Workspace, cache, Data);
+
+        public override DocumentId PrimaryDocumentId
```

Comment:
what does primary here mean?  is this in reference to files that are linked?

Reply:
Renamed.